pkg/migration-tool/planner/planner.go

```diff
@@ -104,11 +106,18 @@ func (c *Config) fetchLastPushedMaxt() (lastPushedMaxt int64, found bool, err er
 	if err != nil {
 		return -1, false, fmt.Errorf("create fetch-last-pushed-maxt reader: %w", err)
 	}
-	result, _, _, err := readClient.Read(context.Background(), query, "")
-	if err != nil {
+	var (
+		doneChan = make(chan *utils.PrompbResponse, 1)
+		errChan  = make(chan error, 1)
+	)
+	readClient.Read(context.Background(), query, 0, "", doneChan, errChan)
+	var ts []*prompb.TimeSeries
+	select {
+	case resp := <-doneChan:
+		ts = resp.Result.Timeseries
+	case err := <-errChan:
```

Comment:
This seems silly. Why dont we have the client have two functions Read and ReadChannels where the latter responds on channels while the former has regular returns. In fact it's easy to write ReadChannels by calling Read.